packages/terra-hookshot/src/HookshotContent.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import onClickOutside from 'react-onclickoutside';
+import styles from './HookshotContent.scss';
+
+const cx = classNames.bind(styles);
+/**
+ * Key code value for the escape key.
+ */
+const KEYCODES = {
+  ESCAPE: 27,
+};
+
+const propTypes = {
+  /**
+   * The children to be displayed as content within the content.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * The function callback when am escape keydown event occurs..
+   */
+  onEsc: PropTypes.func,
+  /**
+   * The function callback when a click outside event occurs.
+   */
+  onOutsideClick: PropTypes.func,
+  /**
+   * The function callback when a resize event occurs.
+   */
+  onResize: PropTypes.func,
+  /**
+   * The function returning the frame html reference.
+   */
+  refCallback: PropTypes.func,
+};
+
+class HookshotContent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClickOutside = this.handleClickOutside.bind(this);
+    this.handleKeydown = this.handleKeydown.bind(this);
+    this.handleResize = this.debounce(this.handleResize.bind(this), 100);
+  }
+
+  componentDidMount() {
+    if (this.props.onEsc) {
```

Comment:
Is it possible for a HookshotContent instance to be mounted initially without onEsc/onResize but later receive them through a `componentWillRecieveProps` call? The event listeners would not be added in that case.